Oracle Provider Generates Invalid Classes

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### cdktf & Language Versions

<!--- Please make note of the version you are using for the `hashicorp/terraform-cdk` and language bindings you are using.
If you are not running the latest version, please upgrade because your issue may have already been fixed. --->

### Affected Resource(s)

- Oracle Provider
  - `containerengine-node-pool.ts`
  - `data-oci-containerengine-node-pool.ts`

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
--->

```
imports/providers/oci/containerengine-node-pool.ts:451:14 - error TS2416: Property 'nodeMetadata' in type 'ContainerengineNodePool' is not assignable to the same property in base type 'TerraformResource'.
  Type '{ [key: string]: string; } | undefined' is not assignable to type '{ [key: string]: any; }'.
    Type 'undefined' is not assignable to type '{ [key: string]: any; }'.

451   public get nodeMetadata(): { [key: string]: string } | undefined {
                 ~~~~~~~~~~~~


imports/providers/oci/containerengine-node-pool.ts:454:14 - error TS2416: Property 'nodeMetadata' in type 'ContainerengineNodePool' is not assignable to the same property in base type 'TerraformResource'.
  Type '{ [key: string]: string; } | undefined' is not assignable to type '{ [key: string]: any; }'.
    Type 'undefined' is not assignable to type '{ [key: string]: any; }'.

454   public set nodeMetadata(value: { [key: string]: string } | undefined) {
                 ~~~~~~~~~~~~


imports/providers/oci/data-oci-containerengine-node-pool.ts:388:10 - error TS2426: Class 'TerraformDataSource' defines instance member accessor 'nodeMetadata', but extended class 'DataOciContainerengineNodePool' defines it as instance member function.

388   public nodeMetadata(key: string): string {
             ~~~~~~~~~~~~


main.ts:3:23 - error TS6133: 'CoreDedicatedVmHost' is declared but its value is never read.

3 import { OciProvider, CoreDedicatedVmHost } from './imports/providers/oci'
                        ~~~~~~~~~~~~~~~~~~~



Found 4 errors.


non-zero exit code 1
```

### Expected Behavior

It should compile 

### Actual Behavior

The typescript types are invalid an therefore aren't compiling

### Steps to Reproduce

Generate `oci` provider bindings and compile - See here for a ready to go example: https://github.com/skorfmann/cdktf-oracle-example

--- a/.github/workflows/examples.yml
+++ b/.github/workflows/examples.yml
@@ -10,6 +10,9 @@ jobs:
     container:
       image: hashicorp/jsii-terraform
 
+    env:
+      CHECKPOINT_DISABLE: "1"
+
     steps:
       - uses: actions/checkout@v2
       - name: installing dependencies

--- a/packages/cdktf/lib/terraform-data-source.ts
+++ b/packages/cdktf/lib/terraform-data-source.ts
@@ -73,7 +73,7 @@ export class TerraformDataSource extends TerraformElement implements ITerraformR
       this.rawOverrides
     )
 
-    attributes['//'] = this.nodeMetadata
+    attributes['//'] = this.constructNodeMetadata
 
     return {
       data: {

--- a/packages/cdktf/lib/terraform-element.ts
+++ b/packages/cdktf/lib/terraform-element.ts
@@ -56,7 +56,7 @@ export class TerraformElement extends Construct {
     curr[lastKey] = value;
   }
 
-  protected get nodeMetadata(): {[key: string]: any} {
+  protected get constructNodeMetadata(): {[key: string]: any} {
     return {
       metadata: {
         path: this.constructNode.path,

--- a/packages/cdktf/lib/terraform-module.ts
+++ b/packages/cdktf/lib/terraform-module.ts
@@ -37,7 +37,7 @@ export abstract class TerraformModule extends TerraformElement {
       this.rawOverrides
     )
 
-    attributes['//'] = this.nodeMetadata
+    attributes['//'] = this.constructNodeMetadata
 
     return {
       module: {

--- a/packages/cdktf/lib/terraform-resource.ts
+++ b/packages/cdktf/lib/terraform-resource.ts
@@ -108,7 +108,7 @@ export class TerraformResource extends TerraformElement implements ITerraformRes
       this.rawOverrides
     )
 
-    attributes['//'] = this.nodeMetadata
+    attributes['//'] = this.constructNodeMetadata
 
     return {
       resource: {

--- a/test/test-providers/cdktf.json
+++ b/test/test-providers/cdktf.json
@@ -11,6 +11,7 @@
     "vault",
     "nomad",
     "external",
-    "terraform-providers/openstack"
+    "terraform-providers/openstack",
+    "oci"
   ]
 }

